src/arrayMethodJoin.js

```diff
@@ -4,9 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultJoin = '';
+
+    if (this.length === 0) {
+      return resultJoin;
+    }
```

Comment:
```suggestion
```

It's redundant. Because if `this.length === 0`, then loop will not even start:
`for (let i = 0; i < this.length; i++)` will mean `for (let i = 0; i < 0; i++)`
In this case 0 is never less than 0 and no iterations wil be made